openmls/src/ciphersuite/hash_ref.rs

```diff
@@ -58,45 +78,73 @@ pub type KeyPackageRef = HashReference;
 /// This value uniquely identifies a proposal.
 pub type ProposalRef = HashReference;
 
+#[derive(TlsSerialize, TlsSize)]
+struct HashReferenceInput<'a> {
+    label: TlsSliceU8<'a, u8>,
+    value: VLBytes,
+}
+
+/// Compute a new [`ProposalRef`] value for a `value`.
+pub fn make_proposal_ref(
```

Comment:
To avoid some type-trickiness involved in passing different labels during KeyPackageRef/ProposalRef construction, I created two new functions `make_proposal_ref` and `make_key_package_ref` that construct references with the correct labels. I am open to other approaches here.